NuRadioMC/simulation/simulation.py

```diff
@@ -1,85 +1,1031 @@
+        # downsample trace to internal simulation sampling rate (the efieldToVoltageConverter upsamples the trace to
+        # 20 GHz by default to achive a good time resolution when the two signals from the two signal paths are added)
+        channelResampler.run(evt, station, det, sampling_rate=1. / dt)
+
+        if add_noise:
+            max_freq = 0.5 / dt
+            Vrms = {}
+            for channel_id in det.get_channel_ids(station.get_id()):
+                norm = integrated_channel_response[station.get_id()][channel_id]
+                Vrms[channel_id] = Vrms_per_channel[station.get_id()][channel_id] / (norm / max_freq) ** 0.5  # normalize noise level to the bandwidth its generated for
+            channelGenericNoiseAdder.run(evt, station, det, amplitude=Vrms, min_freq=0 * units.MHz,
+                                            max_freq=max_freq, type='rayleigh',
+                                            excluded_channels=noiseless_channels[station.get_id()])
+
+        detector_simulation_filter_amp(evt, station, det)
+
+    if time_logger is not None: time_logger.stop_time('detector response')
+
+def build_dummy_event(sid, det, config):
+    """
+    Builds a dummy event for simulation.
+
+    Parameters
+    ----------
+        sid : int
+            The station ID.
+        det : Detector object
+            The detector description defining all channels.
+        config : dict
+            The NuRadioMC configuration dictionary (from the yaml file)
+
+    Returns:
+        object: The built event object.
+    """
+
+    evt = NuRadioReco.framework.event.Event(0, 0)
+    sim_station = NuRadioReco.framework.sim_station.SimStation(sid)
+    sim_station.set_is_neutrino()  # naming not ideal, but this function defines in-ice emission (compared to in-air emission from air showers)
+
+    dt = 1. / config['sampling_rate']
+    # rescale the number of samples to the internal (higher) sampling rate used in the simulation
+    cid = det.get_channel_ids(sid)[0]
+    n_samples = det.get_number_of_samples(sid, cid) / det.get_sampling_frequency(sid, cid) / dt
+    n_samples = int(np.ceil(n_samples / 2.) * 2)  # round to nearest even integer
+
+    for channel_id in det.get_channel_ids(sid):
+        electric_field = NuRadioReco.framework.electric_field.ElectricField([channel_id],
+                                    det.get_relative_position(sim_station.get_id(), channel_id))
+        trace = np.zeros(n_samples)
+        trace[n_samples // 2] = 100 * units.V  # set a signal that should satisfy any trigger and speedup cuts
+        trace[n_samples // 2 + 1] = -100 * units.V
+        electric_field.set_trace(np.array([np.zeros(n_samples), trace, trace]), 1. / dt)
+        electric_field.set_trace_start_time(0)
+        electric_field[efp.azimuth] = 0
+        electric_field[efp.zenith] = 100 * units.deg
+        electric_field[efp.ray_path_type] = 0
+        sim_station.add_electric_field(electric_field)
```

Comment:
Why is that necessary?

Reply:
so that it passed potential speedup cuts